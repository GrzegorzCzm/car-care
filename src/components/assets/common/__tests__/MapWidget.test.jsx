import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import MapWidgetWithStyles, { MapWidget } from '../MapWidget';


describe('MapWidget', () => {
  const props = {
    position: {
      lat: 51.109372,
      lng: 16.958476,
    },
  };


  it('should renders correctly', () => {
    const propsWithClasses = {
      classes: {
        map: 'map-class',
      },
      ...props,
    };

    const wrapper = shallow(<MapWidget {...propsWithClasses} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should contains map style in classes prop', () => {
    const expectedClasses = { map: 'MapWidget-map-1' };

    const wrapper = shallow(<MapWidgetWithStyles {...props} />);

    expect(wrapper.prop('classes')).toEqual(expectedClasses);
  });
});
