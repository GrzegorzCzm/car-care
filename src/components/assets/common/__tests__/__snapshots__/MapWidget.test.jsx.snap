// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapWidget should renders correctly 1`] = `
<Map
  center={
    Object {
      "lat": 51.109372,
      "lng": 16.958476,
    }
  }
  className="map-class"
  zoom={16}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="&copy <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(Marker))
    icon={
      NewClass {
        "_initHooksCalled": true,
        "options": Object {
          "className": "custom-icon",
          "html": "<svg height=\\"30px\\" viewBox=\\"-96 0 464 464\\" width=\\"30px\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-reactroot=\\"\\"><path d=\\"m272 428c0-19.882812-60.890625-36-136-36s-136 16.117188-136 36 60.890625 36 136 36 136-16.117188 136-36zm0 0\\" fill=\\"#adabac\\"></path><path d=\\"m120 160h32v256c0 8.835938-7.164062 16-16 16s-16-7.164062-16-16zm0 0\\" fill=\\"#494342\\"></path><path d=\\"m232 96c0 53.019531-42.980469 96-96 96s-96-42.980469-96-96 42.980469-96 96-96 96 42.980469 96 96zm0 0\\" fill=\\"#ad2943\\"></path><path d=\\"m200 96c0 35.347656-28.652344 64-64 64s-64-28.652344-64-64 28.652344-64 64-64 64 28.652344 64 64zm0 0\\" fill=\\"#ee3446\\"></path></svg>",
        },
      }
    }
    position={
      Object {
        "lat": 51.109372,
        "lng": 16.958476,
      }
    }
  />
</Map>
`;
